Added link to NovoEd from application dashboard

gsidebo opened the pull request:
#### What are the relevant tickets?
Closes #1031 

#### What's this PR do?
Adds a link from Bootcamps to NovoEd if a course is served through NovoEd and they are enrolled

#### How should this be manually tested?
- Add `NOVOED_SAML_LOGIN_URL=https://app.novoed.com/saml/sso?provider=mitbootcamps` to your `.env`
- Make sure you have an application started for some bootcamp run
- Add a `BootcampRunEnrollment` record for some user, then expand the detail section of that run in your app dashboard. You should see a "Start Bootcamp" link at the bottom going to that SAML URL above

#### Any background context you want to provide?
- **These "start bootcamp" links won't work locally.** That app.novoed link is the login link on production, which is already known to work.
- It's OK if there are unchecked bubbles above the "Bootcamp Starts" section. We have some scenarios where users have an enrollment despite not having completed all of the standard application steps. We will correct this UI wonkiness in a separate issue.
- This link will bring users to their NovoEd dashboard, which has a link to the enrolled bootcamp. Ideally we would link our users directly to the bootcamp in NovoEd, but I need to hear back from NovoEd about how we can do that. This dashboard link is sufficient in the meantime.

![ss 2020-10-06 at 09 01 07 ](https://user-images.githubusercontent.com/14932219/95226493-02b36980-07cb-11eb-8afb-e31e4d64494b.png)

mbertrand commented on main/settings.py:
```diff
@@ -872,6 +872,16 @@ def get_all_config_keys():
 NOVOED_API_BASE_URL = get_string(
     "NOVOED_API_BASE_URL", None, description="The base URL of the NovoEd API"
 )
+NOVOED_MIT_BASE_URL = get_string(
```
This doesn't seem to be used for anything?

gsidebo replied:
Thought I was going to need this initially. Removed 👍

mbertrand commented on static/js/pages/applications/ApplicationDashboardPage.js:
```diff
@@ -426,12 +427,27 @@ export class ApplicationDashboardPage extends React.Component<Props, State> {
+    const bootcampStartRow = isNovoEdCourse ? (
+      <BootcampStart
+        ready={novoEdEnrolled}
+        fulfilled={novoEdEnrolled}
```
Looks like `ready` and `fulfilled` will always be the same, so one of them could be removed as a prop.  Or is there a future condition/use-case where that might not be true?

gsidebo replied:
Yup they'll always be the same (for now), but the rest of the detail sections follow a pattern where they receive those two props and they can be different values. To me it makes sense to keep the props consistent even if at the moment they will always be given the same value

gsidebo commented:
@mbertrand Those annoying `caniuse-lite is outdated` warnings finally turned into an error. I fixed that in [this PR](https://github.com/mitodl/bootcamp-ecommerce/pull/1042) and rebased this PR on it. If you have an extra minute, you could review that PR as well

mbertrand reviewed (approved):
👍
